server/jetstream_cluster.go

```diff
@@ -6610,13 +6641,23 @@ RETRY:
 			notActive.Reset(getActivityInterval())
 
 			mrecs := msgsQ.pop()
+
+			// Send acks first for longer RTT situations.
+			for _, mreci := range mrecs {
```

Comment:
Drawback with that is that remote my send the next batch although we may be failing in middle of a batch (in code below), but arguably, this would clearly not be the happy path!

Reply:
I was planning on proposing that the other side does "wait" (with a max wait) for say 1/2 of the previous batch size before sending new batch: basically the sub that updates the outm would have a chance to execute before the next batch is sent allowing to send more at a time. But maybe this approach here is as good (and much simpler).

Reply:
I plan on testing on NGS when this lands, so we will see. Might need more tweaking.